guides/warning-for-head-and-tail.md

```diff
@@ -0,0 +1,63 @@
+# What
+
+CLC has approved the proposal to
+[add `{-# WARNING #-}` to `Data.List.{head,tail}`](https://github.com/haskell/core-libraries-committee/issues/87). This means that the existing users of these functions will receive a compile-time warning about their partiality.
+
+# When
+
+At the very earliest, the change may appear in GHC 9.6 (~ Q1 2023), but more likely GHC 9.8 (~ Q4 2023).
+
+# How
+
+This is not a breaking change, unless a package forces `-Werror`.
+
+If you don't want to change any Haskell code and want to carry on with using `Data.List.{head,tail}`:
+
+1. If you use `-Werror`, set `-Wwarn=warnings-deprecations` to downgrade this area of compiler messages from errors to warnings, so that your build can succeed.
+
+2. If you are annoyed by warnings, set `-Wno-warnings-deprecations` to quash this area of compiler messages. You can also put `:set -Wno-warnings-deprecations` into `.ghci` config file.
+   Note that this also disables any other custom warnings and deprecation warnings!
+
+If you are happy to change code to eliminate warning messages:
+
+1. The easiest way to migrate away from `tail` is to replace it with `drop 1`. This is a drop-in replacement unless you are relying on `tail`'s partiality.
+
+2. Instead of `head` you can use explicit pattern matching: replace any `head foo` with
+
+    ```haskell
+    case foo of
+      [] -> error "<explain, why it is impossible to encounter an empty list here>"
+      hd : _ -> hd
+    ```
+
+3. Another option is to use [`Data.List.uncons`](https://hackage.haskell.org/package/base/docs/Data-List.html#v:uncons): replace any `head foo` with
+
+    ```haskell
+    fromMaybe (error "...") fst (uncons foo)
```

Comment:
Doesn't this need to be something like below instead? As written this doesn't seem to typecheck.

```suggestion
    (fromMaybe (error "...") (fst <$> uncons foo))
```

Reply:
Whoops, thanks, I meant `maybe` instead of `fromMaybe`.